core/src/main/java/org/infinispan/manager/DefaultCacheManager.java

```diff
@@ -470,7 +471,7 @@ public EmbeddedCacheManager startCaches(final String... cacheNames) {
       authzHelper.checkPermission(AuthorizationPermission.LIFECYCLE);
       List<Thread> threads = new ArrayList<Thread>(cacheNames.length);
       final AtomicReference<RuntimeException> exception = new AtomicReference<RuntimeException>(null);
-      for (final String cacheName : cacheNames) {
+      for (final String cacheName : new HashSet<String>(Arrays.asList(cacheNames))) {
```

Comment:
That would still be a HashSet, and all elements would end up in it eventually. We save just the creation of the Arrays.asList wrapper (that's a non copy collection btw) and an Iterator. Not good enough. I'd rather replace  ```List<Thread>``` with ```HashMap<String, Thread>```. The final bill is still cheaper, because HashSet is actually a HashMap on the inside anyway, and we get rid of the ArrayList.